Code/GraphMol/Subgraphs/SubgraphUtils.cpp

```diff
@@ -62,9 +66,9 @@ ROMol *pathToSubmol(const ROMol &mol, const PATH_TYPE &path, bool useQuery,
       subMol->addBond(bond, true);
     }
   } else {
-    for (pathIter = path.begin(); pathIter != path.end(); ++pathIter) {
+    for(auto bondidx : sorted_path) {
       Bond *bond;
-      bond = mol.getBondWithIdx(*pathIter)->copy();
+      bond = mol.getBondWithIdx(bondidx)->copy();
```

Comment:
Might as well delete the line above and replace this with:
```suggestion
      auto bond = mol.getBondWithIdx(bondidx)->copy();
```